src/component/feature/FeatureStrategy/FeatureStrategyForm/FeatureStrategyForm.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import { IFeatureStrategy } from 'interfaces/strategy';
+import {
+    getFeatureStrategyIcon,
+    getHumanReadableStrategyName,
+} from 'utils/strategy-names';
+import { FeatureStrategyType } from '../FeatureStrategyType/FeatureStrategyType';
+import { FeatureStrategyRemove } from '../FeatureStrategyRemove/FeatureStrategyRemove';
+import { FeatureStrategyEnabled } from '../FeatureStrategyEnabled/FeatureStrategyEnabled';
+import { FeatureStrategyConstraints } from '../FeatureStrategyConstraints/FeatureStrategyConstraints';
+import { Button } from '@material-ui/core';
+import {
+    FeatureStrategyProdGuard,
+    useFeatureStrategyProdGuard,
+} from '../FeatureStrategyProdGuard/FeatureStrategyProdGuard';
+import { IFeatureToggle } from 'interfaces/featureToggle';
+import { useStyles } from './FeatureStrategyForm.styles';
+import { formatFeaturePath } from '../FeatureStrategyEdit/FeatureStrategyEdit';
+import { useHistory } from 'react-router-dom';
+import useUiConfig from '../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import ConditionallyRender from '../../../common/ConditionallyRender';
+import { C } from '../../../common/flags';
+import { STRATEGY_FORM_SUBMIT_ID } from 'testIds';
+import { FeatureStrategyConstraints2 } from 'component/feature/FeatureStrategy/FeatureStrategyConstraints2/FeatureStrategyConstraints2';
+import { useConstraintsValidation } from 'component/feature/FeatureStrategy/FeatureStrategyConstraints2/useConstraintsValidation';
+
+interface IFeatureStrategyFormProps {
+    feature: IFeatureToggle;
+    environmentId: string;
+    onSubmit: () => void;
+    loading: boolean;
+    hasAccess: boolean;
+    mode: FeatureStrategyFormMode;
+    strategy: Partial<IFeatureStrategy>;
+    setStrategy: React.Dispatch<
+        React.SetStateAction<Partial<IFeatureStrategy>>
+    >;
+}
+
+export type FeatureStrategyFormMode = 'create' | 'edit';
+
+export const FeatureStrategyForm = ({
+    feature,
+    strategy,
+    setStrategy,
+    environmentId,
+    onSubmit,
+    loading,
+    hasAccess,
+    mode,
+}: IFeatureStrategyFormProps) => {
+    const styles = useStyles();
+    const [showProdGuard, setShowProdGuard] = useState(false);
+    const enableProdGuard = useFeatureStrategyProdGuard(feature, environmentId);
+    const StrategyIcon = getFeatureStrategyIcon(strategy.name ?? '');
```

Comment:
Hmm. I think this function is returning the wrong icons. We added a different icon for gradual rollout a while back
<img width="898" alt="Skjermbilde 2022-03-08 kl  14 47 27" src="https://user-images.githubusercontent.com/16081982/157250552-1515df25-09f0-43c4-8d05-1579008c9e99.png">
.